compaction_test.py

```diff
@@ -361,7 +364,10 @@ def test_large_compaction_warning(self):
         node.nodetool('compact ks large')
         verb = 'Writing' if self.cluster.version() > '2.2' else 'Compacting'
         sizematcher = '\d+ bytes' if self.cluster.version() < LooseVersion('3.6') else '\d+\.\d{3}(K|M|G)iB'
-        node.watch_log_for('{} large partition ks/large:user \({}'.format(verb, sizematcher), from_mark=mark, timeout=180)
+        if cluster.version() >= '4.0':
```

Comment:
nitpicking:
in general, when code is written this way you cannot test different aspects separately:
1. whether the log message you are looking for is the correct one
2. whether the log message is searched for
Here it doesn't matter, but suppose that you have some production code, logic, with the same structure. And suppose that instead of looking for a log you have some expensive operation, say, executing something on a bunch of files, and, instead of log pattern you have an operation to perform on files.

You would like to test each of the two operations separately and application to files separately. Ergo it is more beneficial to extract common part and have a structure like:

```
operation = get_operation_with_some_if_or_something() # tested separately
files.perform(operation)
```
so, getting back to our code at hand:

```
log_message = cluster.version() >= '4.0' and "Detected partition 'use ...." 
                                                                 or "{} large partition .... ".format(...)
node.watch_log_for(log_messagem ....)                            
```